tests/mapped_struct.py

```diff
@@ -591,3 +593,40 @@ class MappedStringMappingBigTest(MappedStringMappingTest):
 class MappedString32MappingBigTest(MappedString32MappingTest):
     TEST_KEYS = [ "%d%s" % (i,k) for k in MappedStringMappingTest.TEST_KEYS for i in xrange(64) ]
     TEST_VALUES = [ v for v in MappedStringMappingTest.TEST_VALUES for i in xrange(64) ]
+
+class BsearchTest(unittest.TestCase):
+    SUPPORTED_DTYPES = [ numpy.uint32, numpy.int32, numpy.uint64, numpy.int64, 
+        numpy.double, numpy.single, numpy.float64, numpy.float32 ]
+
+    UNSUPPORTED_DTYPES = [ numpy.uint16, numpy.int16, numpy.uint8, numpy.int8 ]
+
+    for dtype in SUPPORTED_DTYPES:
+        def testBsearch(self, dtype=dtype):
+            testarray = range(100)
+            random.shuffle(testarray)
+            a = numpy.array(testarray[:50], dtype)
+            b = numpy.array(testarray[50:], dtype)
+            a = numpy.sort(a)
+            for x in a:
+                ix = mapped_struct.bsearch(a, x)
+                self.assertLess(ix, len(a))
+                self.assertEqual(a[ix], x)
+                self.assertTrue(mapped_struct.sorted_contains(a, x))
+            for x in b:
+                ix = mapped_struct.bsearch(a, x)
+                self.assertTrue(ix >= len(a) or a[ix] != x)
+                self.assertTrue(not mapped_struct.sorted_contains(a, x))
```

Comment:
It seems: https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertFalse

It also exists on Python 3